WordPress/Classes/ViewRelated/Post/Preview/PreviewWebKitViewController.swift

```diff
@@ -0,0 +1,127 @@
+        return publishButton
+    }()
+
+    /// Creates a view controller displaying a preview web view.
+    /// - Parameters:
+    ///   - post: The post to use for generating the preview URL and authenticating to the blog. **NOTE**: `previewURL` will be used as the URL instead, when available.
+    ///   - previewURL: The URL to display in the preview web view.
+    init(post: AbstractPost, previewURL: URL? = nil) {
+
+        self.post = post
+
+        let autoUploadInteractor = PostAutoUploadInteractor()
+
+        let isNotCancelableWithFailedToUploadChanges: Bool = post.isFailed && post.hasLocalChanges() && !autoUploadInteractor.canCancelAutoUpload(of: post)
+        canPublish = post.isDraft() || isNotCancelableWithFailedToUploadChanges
+
+        guard let url = PreviewNonceHandler.nonceURL(post: post, previewURL: previewURL) else {
+            super.init(configuration: WebViewControllerConfiguration(url: URL(string: "about:blank")!))
+            return
+        }
+
+        let isPage = post is Page
+
+        let configuration = WebViewControllerConfiguration(url: url)
+        configuration.linkBehavior = isPage ? .hostOnly(url) : .urlOnly(url)
+        configuration.authenticate(blog: post.blog)
+        super.init(configuration: configuration)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        if webView.url?.absoluteString == "about:blank" {
+            showNoResults(withTitle: "No Preview URL available")
+        }
+    }
+
+    override func configureToolbarButtons() {
+        super.configureToolbarButtons()
+
+        let items = toolbarItems(linkBehavior: linkBehavior)
+
+        setToolbarItems(items, animated: false)
+    }
+
+    func toolbarItems(linkBehavior: LinkBehavior) -> [UIBarButtonItem] {
+        let space = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+
+        let items: [UIBarButtonItem]
+
+        switch linkBehavior {
+        case .all, .hostOnly:
+            items = [
+                backButton,
+                space,
+                forwardButton,
+                space,
+                shareButton,
+                space,
+                safariButton
+            ]
+        case .urlOnly:
+            if canPublish {
+                items = [publishButton]
+            } else {
+                items = [shareButton, space, safariButton]
+            }
+        }
+
+        return items
+    }
+
+    @objc private func publishButtonPressed(_ sender: UIBarButtonItem) {
+        PostCoordinator.shared.publish(post)
+        dismiss(animated: true, completion: nil)
+    }
+
+    private func showNoResults(withTitle title: String) {
+        let controller = NoResultsViewController.controllerWith(title: title,
+                                                                buttonTitle: nil,
```

Comment:
Since all these params are optional and default to `nil`, this could just be:
`NoResultsViewController.controllerWith(title: title)`